sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient_private.cs

```diff
@@ -21,13 +19,16 @@ public partial class ConfigurationClient
 
         private static async Task<Response<ConfigurationSetting>> CreateResponseAsync(Response response, CancellationToken cancellation)
         {
-            ConfigurationSetting result = await ConfigurationServiceSerializer.DeserializeSettingAsync(response.ContentStream, cancellation).ConfigureAwait(false);
+            using var contentStream = response.Content.ToStream();
```

Comment:
One observation is that this creates a new MemoryStream under the hood.  Depending on how concerned we are with performance in this scenario, we could update the DeserializeSetting method to take a BinaryData.  If it's just using it to parse a JsonDocument, it should be a small change, and if you change the method signature, you might catch other instances of this bug while you're in there.

Reply:
I've got some time to play with this one.   I'll poke around a bit and see how much churn moving to `BinaryData` would cause.